### ISSUE TITLE
Citation hover (Tooltip) improvements

### ISSUE THREAD

**Issue by geealbers** [gettypubs/quire#152]
For the hover-over tooltips we use with the `q-cite` shortcode (and that I will use elsewhere in other books, like as a glossary), users need to be able to mouse to the text in them and either copy it all, or click a given link. Currently if you mouse toward it you also mouse off the link target and so it disappears before you can get there. Maybe the tip itself is also a hover target and we can extend a transparent area below it to close the gap?

Also, let's have it display on :focus as well as on :hover.

<img width="641" alt="screen shot 2019-03-01 at 3 57 55 pm" src="https://user-images.githubusercontent.com/7796401/53673174-e3417b80-3c3a-11e9-907b-f18e6fc175d4.png">

**Comment by geealbers** [comment]
Rather than have a box that appears on hover, the box should appear on click and then disappear on a second click. And ideally that second click could be on the original target OR anywhere else on the screen but off the box.

A nice implementation I’ve seen of this are the pink + circles sometimes used on Kottke.org for footnotes (like https://kottke.org/19/05/micromobility-will-transform-cities). 

<img width="821" alt="Screen Shot 2019-05-21 at 11 10 27 AM" src="https://user-images.githubusercontent.com/7796401/58123365-83ff6180-7bc0-11e9-9e28-14edd87486ab.png">

Those are footnote specific though, and powered by bigfoot.js which is jQuery-based, so probably not for us. And I don’t think ours need to look the same, but the click action and slight animation are nice, and critically, the box never expands beyond the edges of the main text column, which is an issue with our current hover implementation.

<img width="586" alt="Screen Shot 2019-05-21 at 11 09 46 AM" src="https://user-images.githubusercontent.com/7796401/58123392-8feb2380-7bc0-11e9-8554-74fbbc833b36.png">

**Web Accessible**

Another important feature for us is that these should be fully accessible. That means they should be `<button>`s and use aria expanded and aria collapsed attributes appropriately. There may be some other tips here: https://inclusive-components.design/tooltips-toggletips/.

**Progressively Enhanced**

One *nice* thing about our current implementation is that because they're not JS, the feature works the same in browsers with JS disabled. Also in our current implementation, a citation that looks like "(Evans 1938)" normally, defaults to this when the CSS isn't loaded for any reason: "(Citation: Evans 1938Evans, Walker. American Photographs. New York: Museum of Modern Art, 1938.)" Not perfect, but pretty good. Our new implementation should be progressively enhanced and have a useful default state when either JS or CSS are not loaded. Perhaps the ideal would be: "(Evans 1938 :: Evans, Walker. American Photographs. New York: Museum of Modern Art, 1938.)"

**Toggle Between Two Styles**

For our projects, I can see that we might sometimes want to use an icon like on Kottke, and might sometimes instead want the text itself to be the button, as we do now. So I'd suggest we establish a new param in config.yml to toggle between these two options. It could be `citationPopupStyle` with values of either `text` or `icon`, and with the shortcode defaulting to `text` if the param is not defined or doesn’t match one of the two options.

So with the shortocde `{{< q-cite "Evans 1938" >}}` and `citationPopupStyle: text`, the markup would be structured something like:

```html
<span>
  <button>Evans 1938</button>
  <span>Evans, Walker. American Photographs. New York: Museum of Modern Art, 1938.</span>
</span>
```

Whereas with `citationPopupStyle: icon`, the markup would be:

```html
<span>
  Evans 1938
  <button> + </button>
  <span">Evans, Walker. American Photographs. New York: Museum of Modern Art, 1938.</span>
</span>
```

**Add a Link Inside the Box**

Finally, in the current implementation, you can hover over the link to see the citation, OR click the link to scroll to the bibliography at the bottom. With this new version we’ll loose that ability to click to scroll, so I’d like to instead add it as a default link that always appears in the bottom right corner of the box when it’s open (as in the amazing drawing below), and always just links to the main bibliography section at `#bibliography` rather than the individual entry in the biblio. The text for the link should be defined in a param in config.yml so that users can change it: `citationPopupLinkText: "Bibliography"`

![IMG_7475](https://user-images.githubusercontent.com/7796401/58123980-ec027780-7bc1-11e9-9f5a-5908f3f050da.jpg)

**Comment by geealbers** [comment]
Implemented by @mantasandri with [7eb8b48](https://github.com/gettypubs/quire-starter-theme/commit/7eb8b483eadc5d75a4bfde324f14136fa4a320b2).
